File: src/1331-Rank-Transform-of-an-Array/1331.js
var arrayRankTransform = function(arr) {
    let arrs = arr.slice();
    arrs.sort((a, b) => a - b);
    let st = new Map(), rank = 1;
    
    for (let i = 0; i < arrs.length; i++) {
        if (!st.has(arrs[i])) {
            st.set(arrs[i], rank++);
        }
    }
    
    for (let i = 0; i < arr.length; i++) {
        arr[i] = st.get(arr[i]);
    }
    return arr;
};

File: src/1331-Rank-Transform-of-an-Array/1331.go
func arrayRankTransform(arr []int) []int {
    arrs := make([]int, len(arr))
    copy(arrs, arr)
    sort.Ints(arrs)
    st := make(map[int]int)
    
    for i := 0; i < len(arrs); i++ {
        if _, ok := st[arrs[i]]; !ok {
            st[arrs[i]] = len(st) + 1
        }
    }
    
    for i := 0; i < len(arr); i++ {
        arr[i] = st[arr[i]]
    }
    return arr
}